### ISSUE TITLE
add SSH key verification option

### ISSUE THREAD

**Issue by aaronpk** [aaronpk/IndieAuth.com#97]
http://superuser.com/a/308130

requested by @jerrysievert

**Comment by valpackett** [comment]
https://vtllf.org/blog/ssh-web-sign-in.html

**Comment by aaronpk** [comment]
That URL apparently changed to https://vtllf.org/sshweb

**Comment by singpolyma** [comment]
Could do it using the signing method linked, or (easier) provide ssh login to a space with the key that is set to only execute a command that triggers continuance.  Provide an `ssh://` URI and/or instructions on running `ssh user@ssh.indieauth.com -- login` and after running such the login would proceed.

**Comment by aaronpk** [comment]
Yeah I re-read that post and I'm not quite sure how to make that flow work here.

**Comment by singpolyma** [comment]
The signing-based or the login-based flow? Where are you hosted right now?

**Comment by aaronpk** [comment]
indieauth.com runs on three servers that don't share any state between them, so I'd probably run the SSH agent on only a single server (similar to how openid.indieauth.com works). The main server is a linode VPS.

I'd be happy to take suggestions on how an expected flow would work with SSH verification.

**Comment by singpolyma** [comment]
So, the login-based on would involve ssh keys set up to only be allowed to run a single command (this is a standard sshd feature of the authorixed_keys file) so I would probably be a command run via such a key that would set something in the DB to mean "yes, approve login" and then the web frontend would just spin until the backend saw that value set (or similar) and then approve login (or timeout after some time)
